Github action failed.

Run apt-get update -qq && apt-get install -y -qq postgresql-client > /dev/null
E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
E: Unable to lock directory /var/lib/apt/lists/
W: Problem unlinking the file /var/cache/apt/pkgcache.bin - RemoveCaches (13: Permission denied)
W: Problem unlinking the file /var/cache/apt/srcpkgcache.bin - RemoveCaches (13: Permission denied)
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
	Is the server running locally and accepting connections on that socket?
Error: Process completed with exit code 2.

--- a/.github/workflows/neon-keepalive.yml
+++ b/.github/workflows/neon-keepalive.yml
@@ -18,7 +18,19 @@ jobs:
   keepalive:
     runs-on: ubuntu-latest
     steps:
-      - name: Ping Neon database
+      - name: Check for database URL
         run: |
-          sudo apt-get update -qq && sudo apt-get install -y -qq postgresql-client > /dev/null
-          psql "${{ secrets.NEON_DATABASE_URL }}" -c "SELECT 1 AS ok;" --no-password
+          if [ -z "$NEON_DATABASE_URL" ]; then
+            echo "::error::NEON_DATABASE_URL secret is not set. Please add it under Settings → Secrets and variables → Actions."
+            exit 1
+          fi
+        env:
+          NEON_DATABASE_URL: ${{ secrets.NEON_DATABASE_URL }}
+
+      - name: Install PostgreSQL client
+        run: sudo apt-get update -qq && sudo apt-get install -y -qq postgresql-client > /dev/null
+
+      - name: Ping Neon database
+        run: psql "$NEON_DATABASE_URL" -c "SELECT 1 AS ok;" --no-password
+        env:
+          NEON_DATABASE_URL: ${{ secrets.NEON_DATABASE_URL }}

--- a/.github/workflows/neon-keepalive.yml
+++ b/.github/workflows/neon-keepalive.yml
@@ -20,5 +20,5 @@ jobs:
     steps:
       - name: Ping Neon database
         run: |
-          apt-get update -qq && apt-get install -y -qq postgresql-client > /dev/null
+          sudo apt-get update -qq && sudo apt-get install -y -qq postgresql-client > /dev/null
           psql "${{ secrets.NEON_DATABASE_URL }}" -c "SELECT 1 AS ok;" --no-password

